src/Kiota.Builder/Refiners/GoRefiner.cs

```diff
@@ -130,8 +137,179 @@ public override Task Refine(CodeNamespace generatedCode, CancellationToken cance
+            if (!modelNameSpace.Name.Equals(currentNamespace.Name) && !currentNamespace.IsChildOf(modelNameSpace))
+            {
+                // rename the nested class and move it to the models namespace
+                var classNameList = getPathsName(codeClass, codeClass.Name);
+                
+                var newClassName = string.Join(string.Empty,classNameList);
+                if (!codeClass.Name.Equals(newClassName, StringComparison.OrdinalIgnoreCase))
+                {
+                    currentNamespace.RemoveChildElement(codeClass);
+                    codeClass.Name = newClassName;
+                    codeClass.Parent = modelNameSpace;
+                    modelNameSpace.AddClass(codeClass);
+                }
+            }
+        }
+
+        CrawlTree(currentElement, FlattenNestedHierarchy);
+    }
+
+    private void FlattenGoParamsFileNames(CodeElement currentElement)
+    {
+        if (currentElement is CodeProperty currentProp && currentElement.Parent is CodeClass parentClass && parentClass.IsOfKind(CodeClassKind.RequestConfiguration) && currentProp.IsOfKind(CodePropertyKind.QueryParameters))
+        {
+            var nameList = getPathsName(parentClass, currentProp.Type.Name.ToFirstCharacterUpperCase());
+            var newTypeName = string.Join(string.Empty,nameList);
+                        
+            var type = currentProp.Type;
+            type.Name = newTypeName;
+        }
+
+        if (currentElement is CodeMethod codeMethod && codeMethod.IsOfKind(CodeMethodKind.RequestGenerator, CodeMethodKind.RequestExecutor))
+        {
+            foreach (var param in codeMethod.Parameters){
+                if (param.IsOfKind(CodeParameterKind.RequestConfiguration)){
+                    var newTypeName = string.Join(string.Empty,getPathsName(param, param.Type.Name.ToFirstCharacterUpperCase()));
+                    param.Type.Name = newTypeName;
+                    
+                    foreach (var ct  in param.Type.AllTypes)
+                        if(ct.TypeDefinition != null  && !newTypeName.EndsWith(ct.TypeDefinition.Name.ToFirstCharacterUpperCase()))
+                            ct.TypeDefinition.Name = newTypeName;
+                }
+            }
+            
+        }
+
+        CrawlTree(currentElement, FlattenGoParamsFileNames);
+    }
+
+    private List<string> getPathsName(CodeElement codeClass, string fileName, bool removeDuplicate = true)
+    {
+        // update the code class name to include the entire path
+        var currentNamespace = codeClass.GetImmediateParentOfType<CodeNamespace>();
+        var namespacePathSegments = new List<string>(currentNamespace.Name
+            .Replace(_configuration.ClientNamespaceName, string.Empty)
+            .TrimStart('.')
+            .Split('.'));
+        // add the namespace to the code class Name
+        namespacePathSegments = namespacePathSegments.Where(x => !string.IsNullOrEmpty(x))
```

Comment:
the where can be applied after the split directly to avoid unecessary affectations